modules/gis/src/lib/binary-to-geojson.ts

```diff
@@ -139,18 +173,42 @@ function parseFeatureCollection(dataArray): Feature[] {
         // eslint-disable-next-line no-continue
         continue;
       }
-
-      features.push(parseFeature(data, lastIndex, i));
+      const globalFeatureId = data.globalFeatureIds.value[lastIndex];
+      featuresById.set(globalFeatureId, parseFeature(data, lastIndex, i));
       lastIndex = i;
       lastValue = currValue;
+      count++;
     }
 
     // Last feature
-    features.push(parseFeature(data, lastIndex, data.featureIds.value.length));
+    const globalFeatureId = data.globalFeatureIds.value[lastIndex];
+    featuresById.set(globalFeatureId, parseFeature(data, lastIndex, data.featureIds.value.length));
+    count++;
   }
+
+  // Global ids should be in range [0 - features.length - 1]
+  // as they are assigned sequentially
+  _checkIdRange(featuresById, count);
+  const features: Feature[] = new Array<Feature>(featuresById.size);
+  for (const [featureId, feature] of featuresById) {
+    features[featureId] = feature;
+  }
+
   return features;
 }
 
+function _checkIdRange(featuresById, expectedCount) {
```

Comment:
Nit: we don't use underscore on private function. The absence of `export` already makes this clear.